TCD-2901: Backtrace - Apache and Sourcemap (#2152)
saucelabs/sauce-docs: docs/error-reporting/platform-integrations/source-map.md

---
id: source-map
title: Source Maps Integration Guide - Using backtrace-sourcemap-tools
sidebar_label: Source Maps
description: Configure your JS application to automatically upload sourcemap files.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';

> The following steps guides you through configuring your JS application to automatically upload sourcemap files during project build. Additionally, it will explain how to set the appropriate ID in your Backtrace client to match errors with the corresponding sourcemap files.

Author: ```suggestion
The following steps guide you through configuring your JS application to automatically upload sourcemap files during the project build. Additionally, it explains how to set the appropriate ID in your Backtrace client to match errors with the corresponding sourcemap files.
```

Use the following command to install backtrace-sourcemap-tools as a dev dependency:

```bash
npm install "git@github.com:backtrace-labs/backtrace-sourcemap-tools.git" --save-dev
```

Alternatively, add the following entry to your package.json file under `devDependencies`:

```json
"devDependencies": {
  "backtrace-sourcemap-tools": "git@github.com:backtrace-labs/backtrace-sourcemap-tools.git"
}
```

## Step 3: Configure Sourcemap Upload URL in `package.json`

Specify the URL, token, and universe where Backtrace sourcemap tools will upload sourcemap files when your project is built. Add the following configuration to your `package.json` file under the top-level key `backtrace`:

```json
"backtrace": {
  "sourcemap": {
    "upload": "https://<universe>.sp.backtrace.io:6098/post?format=sourcemap&token=<token>"
  }
}
```

You need to replace `<universe>` with your universe name, and `<token>` with the symbol upload token generated in your Project Settings under the Symbols section. For on-premise Backtrace servers, replace `<universe>.sp.backtrace.io:6098` with the server and port of your Backtrace instance.

## Step 4: Set Up Automatic Sourcemap Uploads

> This method will upload all artifact .map files.

Author: ```suggestion
This method uploads all artifact .map files.
```

First, append `npm run upload` to the `build` command in your package.json's `scripts` section:

```json
"scripts": {
  "build": "my current build command && npm run upload",
  ...
}
```

> Finally, add the action script `upload` to execute `backtrace-sourcemap` with `package.json` and the build folder as parameters. In this example, both `package.json` and the `dist` folder are assumed to be located in the project root directory:

Author: ```suggestion
Finally, add the action script `upload` to run `backtrace-sourcemap` with `package.json` and the build folder as parameters. In this example, both the `package.json` and the `dist` folder are assumed to be located in the project root directory:
```
